packages/clerk-js/src/ui/components/SignIn/ResetPassword.tsx

```diff
@@ -0,0 +1,93 @@
+import { clerkInvalidFAPIResponse } from '../../../core/errors';
+import { withRedirectToHomeSingleSessionGuard } from '../../common';
+import { useCoreSignIn } from '../../contexts';
+import { Col, descriptors, localizationKeys } from '../../customizables';
+import { Card, CardAlert, Form, Header, useCardState, withCardStateProvider } from '../../elements';
+import { useSupportEmail } from '../../hooks/useSupportEmail';
+import { useRouter } from '../../router';
+import { handleError, useFormControl } from '../../utils';
+
+export const _ResetPassword = () => {
+  const signIn = useCoreSignIn();
+  const card = useCardState();
+  const { navigate } = useRouter();
+  const supportEmail = useSupportEmail();
+
+  const passwordField = useFormControl('password', '', {
+    type: 'password',
+    label: localizationKeys('formFieldLabel__newPassword'),
+    isRequired: true,
+  });
+  const confirmField = useFormControl('confirmPassword', '', {
+    type: 'password',
+    label: localizationKeys('formFieldLabel__confirmPassword'),
+    isRequired: true,
+  });
+
+  const canSubmit = passwordField.value === confirmField.value && passwordField.value.length > 7;
+
+  const validateForm = () => {
+    if (passwordField.value && confirmField.value && passwordField.value !== confirmField.value) {
+      confirmField.setError("Passwords don't match.");
+    } else {
+      confirmField.setError(undefined);
+    }
+  };
+
+  const resetPassword = async () => {
+    passwordField.setError(undefined);
+    confirmField.setError(undefined);
```

Comment:
Since we're using this pattern a lot (`setError(undefined)`), should we add a `clearError()` method?

Reply:
While i see your point, I don't think it adds too much value